### DISCUSSION TITLE
QuickLook not opening AR but "preview" after multiple AR button click with complex url

### DISCUSSION THREAD

**Discussion by ryancrunchi** [google/model-viewer#2314] [Q&A]
- Create a model viewer with ar mode
- Set `ios-src` attribute with a complex url encoded (for custom Apple Pay button to show in AR) like
```
<model-viewer id="viewer" ar camera-controls autoplay reveal="manual"
                          poster="https://developer.apple.com/augmented-reality/quick-look/models/biplane/biplane_2x.jpg"
                          src="https://github.com/google/model-viewer/blob/master/packages/shared-assets/models/Astronaut.glb"
                          ios-src="https://developer.apple.com/augmented-reality/quick-look/models/biplane/toy_biplane.usdz#checkoutTitle=Complex%20title&checkoutSubtitle=complex%20sub%20title&price=%20&allowsContentScaling=true&canonicalWebPageURL=https%3A%2F%2Fwww.google.com%2F&applePayButtonType=pay&callToAction=Call%20to%20action">
</model-viewer>
```

- Click on AR button, it shows AR QuickLook with custom Apple Pay button
- Close QuickLook
- Click on AR button 2, it shows AR QuickLook with custom Apple Pay button
- Close QuickLook 2
- Click on AR button 3
Now it shows QuickLook "preview" of .usdz url, instead of AR directly.

<img src="https://user-images.githubusercontent.com/4698648/115844367-0c778400-a420-11eb-90d2-9fbcb87b46c0.jpeg" width=250 />

### Version

- model-viewer: v1.6.0

### Browser Affected

- [ ] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [x] QuickLook

**Edit :** Updated example with model from apple site

**Edit :** After some testing, this issue is related to model-viewer. By adding a single link tag -without model-viewer-, this behaviour is not happening.
```
<a rel="ar" href="https://developer.apple.com/augmented-reality/quick-look/models/biplane/toy_biplane.usdz#checkoutTitle=Complex%20title&checkoutSubtitle=complex%20sub%20title&price=%20&allowsContentScaling=true&canonicalWebPageURL=https%3A%2F%2Fwww.google.com%2F&applePayButtonType=pay&callToAction=Call%20to%20action">
    <img src="https://developer.apple.com/augmented-reality/quick-look/models/biplane/biplane_2x.jpg" width="200px" />
</a>
```

Remember when testing that the Apple Pay button should show in AR with this example :

<img src="https://user-images.githubusercontent.com/4698648/115845830-9116d200-a421-11eb-8dfa-b267e9a2574e.jpeg" width=250 />

**Edit :** This bug also happens with a custom AR button and explicitly calling `.activateAR()` in js.

**Comment by elalish** [comment]
Yeah, that seems reasonable, or it's just a QuickLook bug. Either way I'd suggest filing it on Apple.

**Reply by ryancrunchi** [reply]
Actually, after thinking about it. It is not resources related because this behavior is not happening when the url has no fragment. Without fragment we can open and close QuickLook unlimited number of times

**Comment by ryancrunchi** [comment]
This issue should be reopened. It is related to model-viewer. See the updated initial post

**Reply by elalish** [reply]
Thanks for the detail; can you provide a link where I can repro? Also, I notice your `ios-src` above has a space, which I'm not sure is allowed?

**Reply by ryancrunchi** [reply]
The space is a copy/paste miss, post updated. I don’t have live link atm
